packit_service/worker/jobs.py

```diff
@@ -55,24 +55,75 @@
 logger = logging.getLogger(__name__)
 
 
-def get_handlers_for_event(
-    event: Event, package_config: PackageConfig, packit_comment_command_prefix: str
+def get_handlers_triggered_by_job(
+    event: Event, packit_comment_command_prefix: str
 ) -> Set[Type[JobHandler]]:
     """
-    Get all handlers that we need to run for the given event.
+    Get all handlers that can be triggered by comment (e.g. `/packit build`) or check rerun.
 
-    We need to return all handler classes that:
-    - can react to the given event AND
-    - are configured in the package_config (either directly or as a required job)
+    For comment events we want to get handlers mapped to comment commands. For check rerun event we
+     want to get handlers mapped to check name job.
+     These two sets of handlers are mutually exclusive.
 
-    Examples of the matching can be found in the tests:
-    ./tests/unit/test_jobs.py:test_get_handlers_for_event
+    Args:
+        event: Event which we are reacting to.
+        packit_comment_command_prefix: `/packit` for packit-prod or `/packit-stg` for stg.
 
-    :param event: event which we are reacting to
-    :param package_config: for checking configured jobs
-    :return: set of handler instances that we need to run for given event and user configuration
+    Returns:
+        Set of handlers that are triggered by a comment or check rerun job.
+    """
+    handlers_triggered_by_job = None
+
+    if isinstance(event, AbstractCommentEvent):
+        handlers_triggered_by_job = get_handlers_for_comment(
+            event.comment, packit_comment_command_prefix
+        )
+
+        if handlers_triggered_by_job and not isinstance(
+            event, PullRequestCommentPagureEvent
+        ):
+            event.comment_object.add_reaction(COMMENT_REACTION)
+
+    if isinstance(event, CheckRerunEvent):
+        handlers_triggered_by_job = get_handlers_for_check_rerun(event.check_name_job)
+
+    return handlers_triggered_by_job
+
+
+def handler_matches_to_job(
```

Comment:
Yes, naming is hard..;)

What about this to make it obvious it returns a boolean?

```suggestion
def is_handler_matching_the_event(
```